repo-agent/review-ui/ui/src/UpdateRepo.js

```diff
@@ -0,0 +1,121 @@
+import React, { useState, useEffect } from 'react';
+import yaml from 'js-yaml';
+
+function UpdateRepo({ repo, onCancel, onRepoUpdated }) {
+    const [yamlContent, setYamlContent] = useState('');
+    const [originalYamlContent, setOriginalYamlContent] = useState('');
+    const [isLoading, setIsLoading] = useState(false);
+    const [error, setError] = useState('');
+
+    useEffect(() => {
+        setIsLoading(true);
+        fetch(`/api/repos/${repo.name}/yaml`)
+            .then(res => {
+                if (!res.ok) throw new Error("Failed to fetch repo YAML");
+                return res.json();
+            })
+            .then(data => {
+                if (data.yaml) {
+                    setYamlContent(data.yaml);
+                    setOriginalYamlContent(data.yaml); // Store original YAML content
+                }
+                setIsLoading(false);
+            })
+            .catch(err => {
+                console.error(err);
+                setError(err.message);
+                setIsLoading(false);
+            });
+    }, [repo.name]);
+
+    const handleSubmit = async (e) => {
+        e.preventDefault();
+        setError('');
+        setIsLoading(true);
+
+        try {
+            // Validate YAML
+            const parsed = yaml.load(yamlContent);
+            if (!parsed) throw new Error("YAML is empty or invalid");
+
+            // Extract repoURL from current and original YAML for comparison
+            // Note: The YAML content here is the Spec, so repoURL is at the root.
+            const currentRepoURL = parsed?.repoURL;
+            const originalParsed = yaml.load(originalYamlContent);
+            const originalRepoURL = originalParsed?.repoURL;
+
+            if (currentRepoURL && originalRepoURL && currentRepoURL !== originalRepoURL) {
+                setError('Repository URL cannot be changed in this interface.');
+                setIsLoading(false);
+                return;
+            }
+
+        } catch (e) {
```

Comment:
The `onUpdateComplete()` callback is called in a `finally` block, meaning it will execute regardless of whether the API call succeeded or failed. If the update fails, the UI will incorrectly switch back to the repo list, leading the user to believe their changes were saved. This should only be called on a successful submission.